### PR TITLE
🎨 pydantic2 migration: fixed unit-tests for dynamic-scheduler

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?
- ⬆️ updated dependencies
- made unit tests green

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by giancarloromeo — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]

File: packages/common-library/src/common_library/pydantic_validators.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import datetime
+
+from pydantic import field_validator
+
+
+def _try_convert_str_to_float_or_return(
```
Comment (sanderegg) [inline packages/common-library/src/common_library/pydantic_validators.py:6]: suggestion: `validate_numeric_string_as_timedelta`

Comment (sanderegg) [reply packages/common-library/src/common_library/pydantic_validators.py:6]: also why is this method private? how do you then use it? Is this convoluted thing below really necessary?

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/api/rest/routes.py [hunk 2]
```diff
@@ -16,4 +16,4 @@ def setup_rest_api(app: FastAPI):
     app.include_router(api_router)
 
     app.add_exception_handler(Exception, handle_errors_as_500)
-    app.add_exception_handler(HTTPException, http_exception_as_json_response)
+    app.add_exception_handler(HTTPException, http_exception_as_json_response)  # type: ignore[arg-type]
```
Comment (sanderegg) [inline services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/api/rest/routes.py:19]: why is the ignore necessary? are you sure you are not ignoring something that changed?

**Review by sanderegg — APPROVED** [review]
thanks!

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/api/rest/routes.py (hunk 2 above)
Comment (GitHK) [reply services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/api/rest/routes.py:19]: fixed together

**Pull request merged**